not-so-smart-contracts/solana/sysvar_account_check/README.md

```diff
@@ -0,0 +1,82 @@
+# Missing Sysvar Account Check
+The sysvar (system account) account is often used while validating access control for restricted functions by confirming that the inputted sysvar account by the user matches up with the expected sysvar account. Without this check in place, any user is capable of passing in their own spoofed sysvar account and in turn bypassing any further authentication associated with it, causing potentially disastrous effects.
```

Comment:
Looking at the description, and the recommendations, it's a bit unclear what we are trying to describe here. 

The recommendations are mostly about using `load_instruction_at_checked` over `load_instruction_at`, so maybe the issue here is more "risky usage of load_instruction_at_checked" ?

Reply:
This issue isn’t specific to `load_instructions_at`, and the same check was added for other functions such as `load_current_index` (see: https://docs.rs/solana-program/1.9.2/solana_program/sysvar/instructions/fn.load_current_index.html)

The base issue is not properly validating whether or not an authorized sysvar account has been passed in. I've fixed the recommendations so it's more clear.